engine: 27.5.0 release notes (#21780)
docker/docs: content/manuals/engine/release-notes/27.md

> Release notes for Docker Engine version 27.4 releases.

Reviewer: This  line (and the heading above) still points to 27.4 🙈

Reviewer 2: Oops. fixing!

### 27.5.0

{{< release-date date="2025-01-13" >}}

For a full list of pull requests and changes in this release, refer to the relevant GitHub milestones:

- [docker/cli, 27.5.0 milestone](https://github.com/docker/cli/issues?q=is%3Aclosed+milestone%3A27.5.0)
- [moby/moby, 27.5.0 milestone](https://github.com/moby/moby/issues?q=is%3Aclosed+milestone%3A27.5.0)

#### Bugfixes and enhancements

- containerd image store: Fix passing a build context via tarball to the `/build` endpoint. [moby/moby#49194](https://github.com/moby/moby/pull/49194)

> - Builder GC policies without a `keepStorage` value now inherit the `defaultKeepStorage` limit as intended. [moby/moby#49137](https://github.com/moby/moby/pull/49137)

Reviewer 2: Fixes vale nag

```suggestion
- Builder garbage collection policies without a `keepStorage` value now inherit the `defaultKeepStorage` limit as intended. [moby/moby#49137](https://github.com/moby/moby/pull/49137)
```

- Preserve network labels during daemon startup. [moby/moby#49200](https://github.com/moby/moby/pull/49200)
- Fix a potential race condition error when deleting a container. [moby/moby#49239](https://github.com/moby/moby/pull/49239)

> #### Removed
> 
> - Go SDK: pkg/sysinfo: deprecate NumCPU. This utility has the same behavior as runtime.NumCPU. [moby/moby#49247](https://github.com/moby/moby/pull/49247)
> - Go SDK: pkg/fileutils: deprecate GetTotalUsedFds: this function is only used internally and will be removed in the next release. [moby/moby#49209](https://github.com/moby/moby/pull/49209)
> - Go SDK: pkg/ioutils: deprecate `BytesPipe`, `NewBytesPipe`, `ErrClosed`, `WriteCounter`, `NewWriteCounter`, `NewReaderErrWrapper`, `NopFlusher`, `NopWriter`, `NopWriteCloser`. They were only used internally and will be removed in the next release. [moby/moby#49246](https://github.com/moby/moby/pull/49246), [moby/moby#49255](https://github.com/moby/moby/pull/49255)
> - pkg/reexec: deprecate and migrate to github.com/moby/sys/reexec. This package is deprecated and moved to a separate module. Use github.com/moby/sys/reexec instead. [moby/moby#49135](https://github.com/moby/moby/pull/49135)

Reviewer 2: WDYT about this formatting?

```suggestion
#### Go SDK

- `pkg/sysinfo`: deprecate `NumCPU`. This utility has the same behavior as runtime.NumCPU. [moby/moby#49247](https://github.com/moby/moby/pull/49247)
- `pkg/fileutils`: deprecate `GetTotalUsedFds`: this function is only used internally and will be removed in the next release. [moby/moby#49209](https://github.com/moby/moby/pull/49209)
- `pkg/ioutils`: deprecate `BytesPipe`, `NewBytesPipe`, `ErrClosed`, `WriteCounter`, `NewWriteCounter`, `NewReaderErrWrapper`, `NopFlusher`, `NopWriter`, `NopWriteCloser`. They were only used internally and will be removed in the next release. [moby/moby#49246](https://github.com/moby/moby/pull/49246), [moby/moby#49255](https://github.com/moby/moby/pull/49255)
- `pkg/reexec`: deprecate and migrate to `github.com/moby/sys/reexec`. This package is deprecated and moved to a separate module. Use github.com/moby/sys/reexec instead. [moby/moby#49135](https://github.com/moby/moby/pull/49135)
```

> - `pkg/reexec`: deprecate and migrate to `github.com/moby/sys/reexec`. This package is deprecated and moved to a separate module. Use github.com/moby/sys/reexec instead. [moby/moby#49135](https://github.com/moby/moby/pull/49135)

Reviewer 2: ```suggestion
- `pkg/reexec`: This package is deprecated and moved to a separate module. Use `github.com/moby/sys/reexec` instead. [moby/moby#49135](https://github.com/moby/moby/pull/49135)
```
